api/src/pcapi/core/finance/api.py

```diff
@@ -1973,15 +1986,30 @@ def create_deposit(
         expirationDate=granted_deposit.expiration_date,
     )
 
+    # Edge-cases: Validation of the registration occured over a birthday
+    # Then we need to add recredit to compensate
+    if (
+        eligibility == users_models.EligibilityType.UNDERAGE
+        and _can_be_recredited(beneficiary)
+        and beneficiary.age
+        and age_at_registration
+    ):
+        recredit = do_recredit(beneficiary, deposit)
+        if recredit:
+            # Never happening edge-case: Validation is longer than a year and started when user was 15
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**🎯 suggestion:** ‏
```diff
- # Never happening
+ # Rare 
```

Sinon on dirait que ça ne sert pas :smile: